File: PROGRAMACION II/PRACTICAS_EN_PYTHON/PRACTICA_2_LINEA.py
import matplotlib.pyplot as plt

class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def __str__(self):
        return f'({self.x}, {self.y})'

class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f'Línea desde {self.p1} hasta {self.p2}'

    def dibuja_linea(self):
        print(f'Dibujando línea: {self}')
        plt.plot([self.p1.x, self.p2.x], [self.p1.y, self.p2.y], marker='o', color='b')
        plt.xlim(min(self.p1.x, self.p2.x) - 1, max(self.p1.x, self.p2.x) + 1)
        plt.ylim(min(self.p1.y, self.p2.y) - 1, max(self.p1.y, self.p2.y) + 1)
        plt.grid(True)
        plt.title(f'Línea desde {self.p1} hasta {self.p2}')
        plt.show()

if __name__ == "__main__":
    x1 = float(input("Ingresa la coordenada x1: "))
    y1 = float(input("Ingresa la coordenada y1: "))
    punto1 = Punto(x1, y1)

    x2 = float(input("Ingresa la coordenada x2: "))
    y2 = float(input("Ingresa la coordenada y2: "))
    punto2 = Punto(x2, y2)

    linea = Linea(punto1, punto2)

    print(linea)
    linea.dibuja_linea()

File: PROGRAMACION II/PRACTICAS #1-#2/Practicas 1 y 2 en java/src/main/java/PRACTICA_2/PRACTICA_2_LINEA.java
package PRACTICA_2;

import java.util.Scanner;
import javax.swing.*;
import java.awt.*;

public class PRACTICA_2_LINEA {

    public static class Punto {
        public float x;
        public float y;

        public Punto(float x, float y) {
            this.x = x;
            this.y = y;
        }

        public String toString() {
            return "(" + x + ", " + y + ")";
        }
    }

    public static class Linea {
        public Punto p1;
        public Punto p2;

        public Linea(Punto p1, Punto p2) {
            this.p1 = p1;
            this.p2 = p2;
        }

        public String toString() {
            return "Línea desde " + p1 + " hasta " + p2;
        }

        public void dibujaLinea() {
            System.out.println("Dibujando línea: " + this);
            
            // Crear y mostrar la ventana gráfica para dibujar la línea
            JFrame frame = new JFrame("Línea");
            frame.setSize(400, 400);
            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
            
            JPanel panel = new JPanel() {
                @Override
                protected void paintComponent(Graphics g) {
                    super.paintComponent(g);
                    g.setColor(Color.RED);
                    // Dibujar la línea entre los puntos p1 y p2
                    g.drawLine((int)p1.x, (int)p1.y, (int)p2.x, (int)p2.y);
                }
            };
            
            frame.add(panel);
            frame.setVisible(true);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingresa la coordenada x1: ");
        float x1 = scanner.nextFloat();
        System.out.print("Ingresa la coordenada y1: ");
        float y1 = scanner.nextFloat();
        Punto punto1 = new Punto(x1, y1);

        System.out.print("Ingresa la coordenada x2: ");
        float x2 = scanner.nextFloat();
        System.out.print("Ingresa la coordenada y2: ");
        float y2 = scanner.nextFloat();
        Punto punto2 = new Punto(x2, y2);

        Linea linea = new Linea(punto1, punto2);

        System.out.println(linea.toString());
        linea.dibujaLinea();
    }
}
